src/runtime/node/timers/promises/index.ts

```diff
@@ -0,0 +1,20 @@
+import { notImplementedAsync } from "../../../_internal/utils";
+import type timers from "node:timers/promises";
+import { Scheduler } from "./scheduler";
+
+export const setImmediate: typeof timers.setImmediate = notImplementedAsync(
```

Comment:
I think this makes sense since setImmediate is quite problematic for runtimes to implement so it's native availability is spotty.

We could try to shim it, but it's challenging because setImmediate is something in between queueMicrotask and setTimeout: https://nodejs.org/en/learn/asynchronous-work/event-loop-timers-and-nexttick#setimmediate-vs-settimeout

Some have used `postMessage` to get closer to the real behavior (e.g. [vscode here](https://github.com/microsoft/vscode/blob/e045e3b1b25de81232b7db32747f31bf90d02d13/src/vs/base/common/platform.ts#L193-L193)) but even then there are issues, proving this to be a problematic api to shim: https://github.com/microsoft/vscode/issues/133173

I'm ok throwing for now and leaving a comment in the code about these considerations and come back to it later once the need arises.

Reply:
Maybe we could try a proimisified version of default `setImmediate` and fallback to queueMicrotask? If the end-goal is maximum compatibility, fallback behaviors (even if not precise) IMO would be better than throwing errors

Reply:
queueMicrotask will result in ordering being different in the polyfilled version. a postMessage implementation is likely the closes we can get to the real thing, unless we actually implement this in `workerd` natively.

Reply:
promisified version of setImmediate could be likely used in in libraries to make a delay. Exposing an implementation from native scheduler of workerd runtime would be amazing. In the meantime i suggest to simply promisify the global setImmediate to give basic functionality rather than breaking.

Reply:
we could do that if setImmediate is present, but keep in mind that many runtimes including deno and workerd don't support it.

Reply:
Yeah it would be anyway not consistent with Node.js core and i belive runtime-specific Node polyfills would be different...

I think unenv could simply try it best to make things working on best-effort basis at best.

Checking Deno 1.44.0 it seems to work:

<img width="675" alt="image" src="https://github.com/unjs/unenv/assets/5158436/dbe35300-25d0-42cc-a1fd-91798e0a53c6">

And bun returning promise with stagenly not calling cb or somehow a bug:

<img width="731" alt="image" src="https://github.com/unjs/unenv/assets/5158436/c163c67e-56b9-476d-a6d9-a4377fcee1f6">

Reply:
Let's defer this discusison